x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

```diff
@@ -64,6 +66,9 @@ public static String randomIndexPattern(Feature... features) {
         if (canAdd(Features.CROSS_CLUSTER, features)) {
             var cluster = randomIdentifier();
             pattern = maybeQuote(cluster + ":" + pattern);
+        } else if (EsqlCapabilities.Cap.INDEX_COMPONENT_SELECTORS.isEnabled() && canAdd(Features.INDEX_SELECTOR, features)) {
+            var selector = ESTestCase.randomFrom(IndexComponentSelector.values());
+            pattern = maybeQuote(pattern + "::" + selector.getKey());
```

Comment:
Is it possible to use selectors with remote clusters?
If so, this should be separate `if` branch, not `else if`.

Reply:
The selector feature is not yet supported by remote clusters in CCS, so I made them exclusive of each other, but for testing purposes they can technically be used together. I've updated this logic to allow both parts to be present for coverage.

Reply:
Thank you!
Unless it is enforced by parser, it is better to keep those branches independent

Reply:
Just double checking that it was committed, since this block is not marked as outdated by github.

Reply:
It should be present now - was cleaning up a couple things before pushing

Reply:
Could you please address this request?

Reply:
Could you please address this request?

Reply:
There has been a change since this thread started around how we plan to move forward with selectors and cluster ids. They are no longer compatible with each other in the parser for the time being. Until the selector is supported in CCS, the parser will throw an exception if the two are used together. As such, this is an `else if` block to keep from interfering with other randomized tests. Would you rather it be updated to a separate `if` block and all uses of this function add a `without(INDEX_COMPONENT_SELECTORS)` in the arguments?